### ISSUE TITLE
Different implementation for Model.load and Model._load

### ISSUE THREAD

**Issue by Darejkal** [ultralytics/ultralytics#24522]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Hi, I was training yolo using ultralytics and wondered why the implementations of `Model.load` and `Model._load` are different from each other. To me it seems that these 2 methods can be merged, since some difference may cause troubles during training.

For example,  `_load` currently set `self.ckpt_path`, which is later required during resume training. `load` does not set this property. `load` choose to set only `self.ckpt` and `pretrained` instead, causing training to start from epoch 0 even with `resume=True`.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Darejkal, thank you for your interest in Ultralytics 🚀! This is an automated response to help get your issue triaged quickly, and an Ultralytics engineer will also assist soon.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python) and [CLI](https://docs.ultralytics.com/usage/cli) usage examples and where many of the most common questions may already be answered.

Since your report mentions potentially inconsistent behavior around `Model.load`, `Model._load`, and `resume=True`, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) showing the exact steps, code, model path usage, and logs needed to reproduce the behavior 🔍. This will help us verify whether this is an implementation inconsistency, expected behavior, or a bug.

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes) and [Tasks](https://docs.ultralytics.com/tasks) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by raimbekovm** [comment]
thanks @Darejkal - we put up a fix in #24523, could you install that branch (`pip install git+https://github.com/ultralytics/ultralytics.git@fix/model-load-ckpt-path`) and confirm it resolves the resume issue on your side?

**Comment by Darejkal** [comment]
I think it might be better to call Model._load inside Model.load, but I guess a quick patch for self.ckpt_path is fair. Thanks.

**Comment by raimbekovm** [comment]
agreed the quick patch is the right call - `load()` and `_load()` are intentionally separate: `load()` does name/shape-matched weight transfer into an already-built architecture, while `_load()` swaps the whole model object from the checkpoint. routing `load()` through `_load()` would discard any custom architecture (verified: `YOLO("custom-nc3.yaml").load("yolo11n.pt")` keeps nc=3 today, but via `_load()` it becomes the 80-class yolo11n), breaking the main transfer-learning use case. thanks for the report and the suggestion!
